code/game/objects/structures/crates_lockers/closets/secure/freezer.dm

```diff
@@ -152,8 +152,14 @@
 		return
 	cooldown = 5
 	var/containers = 0
+	var/containers_ready = FALSE
 	for(var/obj/item/reagent_container/glass/I in freezer.contents)
 		if(I.reagents.replace_with(polymerization_recipe, "paraformaldehyde", 3))
 			containers++
+			if(!I.reagents.has_reagent("formaldehyde", 3) || !I.reagents.has_reagent("water", 3))
```

Comment:
What about frost oil?

Reply:
frost oil is used to mix with formaldehyde to instantly create paraformaldehyde, wouldn't make sense to use it in the freezer